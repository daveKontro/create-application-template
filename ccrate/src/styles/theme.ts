const theme = {
  font: 'exo2',
  colors: {
    palette: {
      background: '#182f37',
      primary: '#e2b432',
    },
    get link() { return this.palette.primary },
  },
}

export default theme
