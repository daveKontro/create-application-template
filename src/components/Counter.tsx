import { FC, useState } from 'react'
import { StyledCounter } from '../styles/Counter.styled'

// NOTE update count then edit App.tsx... thanks
// to ReactRefreshWebpackPlugin state is preserved

type Count = number

export const Counter: FC = (): JSX.Element => {
  const [count, setCount] = useState<Count>(0)

  const handleClickCounter = () => setCount((count: Count) => ++count)

  return (
    <div>
      <StyledCounter onClick={handleClickCounter}>
        count:&nbsp;{count}
      </StyledCounter>
    </div>
  )
}
