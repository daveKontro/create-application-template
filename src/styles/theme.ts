const theme = {
  font: 'exo2',
  colors: {
    palette: {
      background: '#454145',
      primary: '#cec2eb',
    },
    get link() { return this.palette.primary },
  },
}

export default theme
