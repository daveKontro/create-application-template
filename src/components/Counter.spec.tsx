import { render, screen } from '@testing-library/react'
import 'jest-styled-components'
import { userEvent } from '@testing-library/user-event'
import { ThemeProvider } from 'styled-components'
import theme from '../styles/theme'
import { Counter } from './Counter'

test('count increases per click', async () => {
  render(
    <ThemeProvider theme={theme}>
      <Counter />
    </ThemeProvider>
  )
  const button = await screen.findByRole('button', { name: /count/i })

  const user = userEvent.setup()
  await user.click(button)
  const results = button.innerHTML.match(/\d/g)

  expect(results).toBeArrayOfSize(1)
  expect(results).toIncludeAllMembers(['1'])
})


test('styled with palette colors', async() => {
  render(
    <ThemeProvider theme={theme}>
      <Counter />
    </ThemeProvider>
  )
  const button = await screen.findByRole('button', { name: /count/i })

  const { primary, background } = theme.colors.palette

  expect(button).toMatchSnapshot()
  expect(button).toHaveStyleRule('color', primary)
  expect(button).toHaveStyleRule('background-color', background)
})
