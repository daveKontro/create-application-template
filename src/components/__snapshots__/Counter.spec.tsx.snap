// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled with palette colors 1`] = `
.c0 {
  height: 40px;
  padding: 10px;
  font-size: 18px;
  color: #cec2eb;
  background-color: #454145;
  border: 1px solid #cec2eb;
  border-radius: 8px;
  transition-duration: 0.5s;
}

.c0:hover {
  border-radius: 20px;
}

<button
  class="c0"
>
  count:Â 
  0
</button>
`;
